### PR TITLE
Modular lambda functions

### PR DESCRIPTION
**Pull request by nashley**
Iteration 2 backend.
Uses multiple lambdas, API Gateways, and a step function to make the process modular and very scalable.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,20 +4,110 @@ node_js:
...
+    role: "arn:aws:iam::957512267502:role/LambdaDeployment"
+    runtime: "nodejs6.10"
+    handler_name: "handler"
+    zip: "${TRAVIS_BUILD_DIR}/server/orders/context_call_v2_orders.zip"
+    access_key_id: $AWSACCESSKEY
+    secret_access_key: $SECRETACCESSKEY
+    environment_variables:
+      - SECRET=$AWS_LAMBDA_ENV_SECRET
+    on:
+      branch: modular_lambda
```
Comment (NikolasMcGovern) [inline .travis.yml]: Change to master

File: .travis.yml [hunk 2]
```diff
@@ -4,20 +4,110 @@ node_js:
...
+    secret_access_key: $SECRETACCESSKEY
+    environment_variables:
+      - SECRET=$AWS_LAMBDA_ENV_SECRET
+      - API_SECRET=$AWS_LAMBDA_ENV_API_SECRET
+      - API_TOKEN=$AWS_LAMBDA_ENV_API_TOKEN
+      - PHONE_NUMBER=$AWS_LAMBDA_ENV_PHONE_NUMBER
+      - USER_ID=$AWS_LAMBDA_ENV_USER_ID
+      - CALLBACK_WAIT_URL=$AWS_LAMBDA_ENV_CALLBACK_WAIT_URL
+    on:
+      branch: modular_lambda
```
Comment (NikolasMcGovern) [inline .travis.yml]: Change to master

File: .travis.yml [hunk 3]
```diff
@@ -4,20 +4,110 @@ node_js:
...
```
Comment (NikolasMcGovern) [inline .travis.yml]: Change to master

File: client/package.json [hunk 4]
```diff
@@ -3,7 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
-    "@bandwidth/shared-components": "^3.0.9",
+    "@bandwidth/shared-components": "3.2.0",
```
Comment (nashley) [inline client/package.json]: Should we get this sorted out first? I'm thinking we leave it for now.

Comment (NikolasMcGovern) [reply client/package.json]: I think leaving it is best for now. We can adjust it when the problem is fixed.

File: server/local_dependencies/simple-bandwidth-api/index.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+	let req = https.request(options, (resp) => {
+		var data = '';
+
+		resp.on('data', (chunk) => {
+			data += chunk;
+		});
+
+		resp.on('end', () => {
+			console.log("Bandwidth response: '" + data + "'");
+			// TODO verify return data/check docs/return the successful action to the client
```
Comment (nashley) [inline server/local_dependencies/simple-bandwidth-api/index.js]: I'm not fond of merging `TODO`s into master, but I guess we can leave this for now.

**Pull request merged**